### PR TITLE
Pokedex Plus fixes

### LINKED ISSUE #4512: "Bugs" with the HGSS Pokedex
### Description

There's 2 types of errors as far as I can tell, but they all deal with the evo page of the dex.
1.is errors loading the target species icons.
-when a species has multiple evo methods to the same target species it reloads the icon.

2. dpad navigation issues when scrolling the evolutions.
    moving up on the dpad works, fine the issue is Down Dpad, doesn't really work quite right.
plus the selector arrow isn't loaded unless more than 1 evo method is checked and you've seen the target species,
which may be by intention but it hides the functionality that you can  scroll and select the evolution to navigate to their pages.

also once you DO have the selector arrow accessed, it doesnt block navigation to a mon that isn't in the dex list/index.
which breaks dex scrolling if you select them, forcing you to back out with B.

https://discord.com/channels/419213663107416084/419214240277200898/1237168041385787543
Link above is to the rhh discord with my full notes on the subject with gifs.

evolution errors

navigation errors

Situation illustrated was
caught wurmple seen cascoon, never saw silcoon etc.

2 evo methods for combusken  and 2 evo methods for blaziken

.evolutions = EVOLUTION({EVO_LEVEL, 16, SPECIES_COMBUSKEN},
                        {EVO_LEVEL_ATK_GT_DEF, 16, SPECIES_COMBUSKEN}),

.evolutions = EVOLUTION({EVO_LEVEL, 36, SPECIES_BLAZIKEN},
                        {EVO_DARK_SCROLL, 16, SPECIES_BLAZIKEN}),

### Version

1.8.3 (Latest release)

### Upcoming/master Version

_No response_

### Discord contact info

_No response_

### PR DESCRIPTION
**Pull request by Sneed69**
## Issue(s) that this PR fixes
#4512

## Images/Videos
image

https://github.com/rh-hideout/pokeemerald-expansion/assets/56992013/043fab55-b268-4bd8-9fab-5eb390225ba4

## **Discord contact info**
duke5614

### CODE REVIEW

**Comment by Sneed69** [comment]
Also restored the missing "has no evolution" text mentioned here: https://github.com/rh-hideout/pokeemerald-expansion/issues/4512#issuecomment-2098797338

**Comment by Meister-anon** [comment]
> Also restored the missing "has no evolution" text mentioned here: [#4512 (comment)](https://github.com/rh-hideout/pokeemerald-expansion/issues/4512#issuecomment-2098797338)

Did you test that in game?
Just putting the print in the null argument alone would cause other issues...

**Comment by Sneed69** [comment]
> > Also restored the missing "has no evolution" text mentioned here: [#4512 (comment)](https://github.com/rh-hideout/pokeemerald-expansion/issues/4512#issuecomment-2098797338)
> 
> Did you test that in game? Just putting the print in the null argument alone would cause other issues...

Yes, I tested it.
Not sure what you mean about causing issues from a "null argument."

**Comment by Meister-anon** [comment]
Pretty much that with the recursive call at the end of the function, it would change the species in evolution and do everything again.

And evolution is null when there is no evo, that applies to mon with no evolution as well as mon that are final evolutions in the line.

So when I did what you did and checked on a mon that had evos, it printed the no evo text on top of all the other text.
as well as printed it for the final mon in the evo line.

**Comment by Sneed69** [comment]
Fixed
image

**Comment by Meister-anon** [comment]
Nice fix, much better than what I came up with.

**Comment by Sneed69** [comment]
Fixed everything in #4512 except https://github.com/rh-hideout/pokeemerald-expansion/issues/4512#issuecomment-2100615174 as that seems like intended behavior.

File: src/pokedex_plus_hgss.c [hunk 1]
```diff
@@ -6431,21 +6429,25 @@ static u8 PrintEvolutionTargetSpeciesAndMethod(u8 taskId, u16 species, u8 depth,
...
+    //If there are no evolutions print text and return
+    if (evolutions == NULL)
+    {
+        if (depth == 0)
+        {
+            StringExpandPlaceholders(gStringVar4, sText_EVO_NONE);
+            PrintInfoScreenTextSmall(gStringVar4, base_x-7-7, base_y + base_y_offset*(*depth_i));
+        }
+        return;
+    }
```
Comment (AlexOn1ine) [inline src/pokedex_plus_hgss.c:6450]: this could be combined into one if statement

**Review by AlexOn1ine — APPROVED** [review]

**Pull request merged**